File: android-room-sample-master/app/src/javaDagger/java/com/emreeran/android/roomsample/ui/NavigationController.java
package com.emreeran.android.roomsample.ui;

import android.support.v4.app.FragmentManager;

import com.emreeran.android.roomsample.MainActivity;
import com.emreeran.android.roomsample.R;

import javax.inject.Inject;

/**
 * Created by Emre Eran on 26.05.2018.
 */
public class NavigationController {
    private FragmentManager mFragmentManager;
    private int mContainerId = R.id.frame;

    @Inject
    public NavigationController(MainActivity mainActivity) {
        mFragmentManager = mainActivity.getSupportFragmentManager();
    }

    public void navigateToInitializer() {
        mFragmentManager.beginTransaction()
                .replace(mContainerId, InitializerFragment.create())
                .commitAllowingStateLoss();
    }

    public void navigateToFeed() {
        mFragmentManager.beginTransaction()
                .setCustomAnimations(android.R.anim.slide_in_left, android.R.anim.slide_out_right)
                .replace(mContainerId, FeedFragment.create())
                .commitAllowingStateLoss();
    }

    public void navigateToPostDetail(int postId) {
        mFragmentManager.beginTransaction()
                .addToBackStack(PostDetailFragment.class.getSimpleName())
                .setCustomAnimations(
                        android.R.anim.slide_in_left, android.R.anim.slide_out_right,
                        android.R.anim.slide_in_left, android.R.anim.slide_out_right)
                .replace(mContainerId, PostDetailFragment.create(postId))
                .commitAllowingStateLoss();
    }
}

File: android-room-sample-master/app/src/kotlinDagger/java/com/emreeran/android/roomsample/ui/NavigationController.kt
package com.emreeran.android.roomsample.ui

import com.emreeran.android.roomsample.R
import com.emreeran.android.roomsample.MainActivity
import javax.inject.Inject


/**
 * Created by Emre Eran on 21.04.2018.
 */
class NavigationController @Inject constructor(activity: MainActivity) {
    private val fragmentManager = activity.supportFragmentManager!!
    private val containerId = R.id.frame

    fun navigateToInitializer() {
        fragmentManager.beginTransaction()
                .replace(containerId, InitializerFragment.create())
                .commitAllowingStateLoss()
    }

    fun navigateToFeed() {
        fragmentManager.beginTransaction()
                .setCustomAnimations(android.R.anim.slide_in_left, android.R.anim.slide_out_right)
                .replace(containerId, FeedFragment.create())
                .commitAllowingStateLoss()
    }

    fun navigateToPostDetail(postId: Int) {
        fragmentManager.beginTransaction()
                .addToBackStack(PostDetailFragment::class.java.simpleName)
                .setCustomAnimations(
                        android.R.anim.slide_in_left, android.R.anim.slide_out_right,
                        android.R.anim.slide_in_left, android.R.anim.slide_out_right)
                .replace(containerId, PostDetailFragment.create(postId))
                .commitAllowingStateLoss()
    }
}